### PR TITLE
chore: Use C++20 concepts unconditionally

### PR DESCRIPTION
**Pull request by paulgessinger**
With C++20 being the minimum standard, we don't need to gate concept
usage behind preprocessor macros.

### CODE REVIEW

File: Core/include/Acts/Utilities/Concepts.hpp [hunk 1]
```diff
@@ -1,26 +0,0 @@
...
-
-#pragma once
-
-#if defined(__cpp_concepts)
-
-#define ACTS_REQUIRES(x) requires(x)
-#define ACTS_CONCEPT(x) x
-#define ACTS_STATIC_CHECK_CONCEPT(check_concept, check_type) \
-  static_assert(check_concept<check_type>,                   \
-                #check_type " does not fulfill " #check_concept)
```
Comment (paulgessinger) [inline Core/include/Acts/Utilities/Concepts.hpp:17]: Should we keep this as a convenience or just use the expanded `static_assert` everywhere?

Comment (andiwand) [reply Core/include/Acts/Utilities/Concepts.hpp:17]: To me the direct use of `static_asset` looks cleaner

**Review by andiwand — APPROVED** [review]
LGTM 👍

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [3e4cb140488f9fd338503ce6b5afb12e2c7ae46a]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/41283529) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7764584)!

**Comment by andiwand** [comment]
Surprisingly this is breaking as Athena binds to `ACTS_STATIC_CHECK_CONCEPT`

**Pull request merged**